### PR TITLE
1.3.24 update from Ledger with Veil support

### PR DESCRIPTION
**Pull request by Kryptoxic**
Updated the app to 1.3.16 and also included Veil support.

## Bounty
Bounty address: bv1ql9t8rzaqqtet5urz2j4ywqn5e38efzxr7yvemc

Basecoin address is due to using an exchange address. I'm currently getting funds together for a new PC because my previous computer died on me so I'm currently slow in pushing out codes. Help me out with a donation to a new PC! :D

### CODE REVIEW

File: src/btchip_apdu_hash_input_start.c [hunk 1]
```diff
@@ -85,7 +85,11 @@ unsigned short btchip_apdu_hash_input_start() {
                 (G_coin_config->kind == COIN_KIND_BITCOIN_CASH ? usingCashAddr
                                                                : 0);
             btchip_context_D.usingOverwinter = 0;
+<<<<<<< HEAD
             if ((G_coin_config->kind == COIN_KIND_ZCASH) || (G_coin_config->kind == COIN_KIND_KOMODO)) {
+=======
+            if ((G_coin_config->kind == COIN_KIND_ZCASH) || (G_coin_config->kind == COIN_KIND_KOMODO) || (G_coin_config->kind == COIN_KIND_ZCLASSIC) || (G_coin_config->kind == COIN_KIND_RESISTANCE)) {
+>>>>>>> upstream/master
```
Comment (CaveSpectre11) [inline src/btchip_apdu_hash_input_start.c]: @Kryptoxic ; you have an unaddressed merge here

**Review by CaveSpectre11 — CHANGES_REQUESTED** [review]
There are a number of files with unaddressed merges in it

File: src/btchip_apdu_hash_input_start.c (hunk 1 above)
Comment (Kryptoxic) [reply src/btchip_apdu_hash_input_start.c]: Apologies for the late reply, I was away and busy for the month of December. I have manually looked through the entire project and merged the remaining files.

**Comment by Kryptoxic** [comment]
Apologies for the late reply, I was away and busy for the month of December. I have manually looked through the entire project and merged the remaining files.

**Comment by Kryptoxic** [comment]
Updated with ledger's code from version 1.3.17

**Comment by Kryptoxic** [comment]
Updated code with upstream changes from ledger

**Pull request closed without merging**